File: Data Structures & Algorithms/rotate-array/submission-2.java
class Solution {
    // public void rotate(int[] nums, int k) {

    //     int n = nums.length;
    //     k %= n;

    //     while (k > 0){
    //         int temp = nums[n - 1];

    //         for (int i = n - 1; i > 0 ; i--){
    //             nums[i] = nums[i - 1];
    //         }

    //         nums[0] = temp;
    //         k--;
    //     }

        
    // }

    public void rotate(int[] nums, int k){
        int n = nums.length;
        k %= n;

        reverse(nums, 0 , n -1);
        reverse(nums,0 , k-1);
        reverse(nums, k, n-1);
    }


    private void reverse(int[] nums, int l, int r){
        while (l < r){
            int temp = nums[l];
            nums[l] = nums[r];
            nums[r] = temp;

            l++;
            r--;
        }
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let freq = {};

        //freq of elements

        for (const num of nums){
            freq[num] = (freq[num] || 0) + 1;
        }

        const minHeap = [];

        for (const xCount of Object.entries(freq)){
            minHeap.push(xCount);
            minHeap.sort((a,b) => b[1] - a[1] );

            if (minHeap.length > k){
                minHeap.pop()
            }
        }

        return minHeap.map(pair => pair[0])

    }
}
